File: backend/src/main/java/com/example/doctor/appointment/entity/Doctor.java
package com.example.doctor.appointment.entity;

import com.example.doctor.appointment.enums.DoctorStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "doctors")
@Builder
public class Doctor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @OneToOne
    @JoinColumn(name = "user_id",nullable = false)
    private User user;
    @ManyToOne
    @JoinColumn(name = "department_id")
    private Department department;
    @OneToMany(mappedBy = "doctor",cascade = CascadeType.ALL,fetch = FetchType.EAGER)
    private List<Schedule> schedules;
    private String doctorNo;
    private String designation;
    private String biography;
    private String address;
    private String city;
    private String state;
    private String zip;
    private Double fee;
    private String signature;
    @Enumerated(EnumType.STRING)
    private DoctorStatus status;
    @OneToMany(mappedBy = "doctor")
    private List<Appointment> appointments;

    @OneToMany(mappedBy = "doctor")
    private List<Prescription> prescriptions;
}

File: frontend/src/types/Doctor.ts
import type { Department } from "./Department"
import type { Schedule } from "./Schedule"
import type { User } from "./User"

export interface Doctor {
    id: any
    departmentId: number
    doctorNo: string
    department: Department
    designation: string
    biography: string
    user: User
    address: string
    city: string
    state: string
    zip: string
    fee: string
    signature: string
    status: unknown
    schedules: Schedule[]
    country: string
    specialization: string[]
    experience: any
    reviews: any[]
}